os-eap-migration/added/launch/openshift-migrate-common.sh

```diff
@@ -0,0 +1,93 @@
+  local terminatingFile="${JBOSS_HOME}/terminatingMigration"
+
+  (runMigrationServer "$instanceDir" "${txOptions}") &
+
+  PID=$!
+
+  rm -f "${terminatingFile}"
+
+  trap "echo Received TERM ; touch \"${terminatingFile}\" ; kill -TERM $PID ; " TERM
+  local success=false
+  local message="Finished, migration pod has been terminated"
+  ${JBOSS_HOME}/bin/readinessProbe.sh
+
+  if [ $? -eq 0 ] ; then
+    echo "$(date): Server started, checking for transactions"
+    local startTime=$(date +'%s')
+    local endTime=$((startTime + ${RECOVERY_TIMEOUT} + 1))
+
+    local socketBinding=$(run_cli_cmd '/subsystem=transactions/:read-attribute(name="socket-binding")' | grep -w result | sed -e 's+^.*=> "++' -e 's+".*$++')
+    local recoveryPort=$(run_cli_cmd '/socket-binding-group=standard-sockets/socket-binding='"${socketBinding}"'/:read-attribute(name="bound-port")' | grep -w result | sed -e 's+^.*=> ++')
+    local recoveryHost=$(run_cli_cmd '/socket-binding-group=standard-sockets/socket-binding='"${socketBinding}"'/:read-attribute(name="bound-address")' | grep -w result | sed -e 's+^.*=> "++' -e 's+".*$++')
+
+    if [ -n "${recoveryPort}" ] ; then
+      local recoveryClass="com.arjuna.ats.arjuna.tools.RecoveryMonitor"
+      recoveryJar=$(find "${JBOSS_HOME}" -name \*.jar | xargs grep -l "${recoveryClass}")
+      if [ -n "${recoveryJar}" ] ; then
+        echo "$(date): Executing synchronous recovery scan"
+        java -cp "${recoveryJar}" "${recoveryClass}" -host "${recoveryHost}" -port "${recoveryPort}"
+      fi
+    fi
+
+    while [ $(date +'%s') -lt $endTime -a ! -f "${terminatingFile}" ] ; do
+      run_cli_cmd '/subsystem=transactions/log-store=log-store/:probe' > /dev/null 2>&1
+      local transactions="$(run_cli_cmd 'ls /subsystem=transactions/log-store=log-store/transactions')"
+      if [ -z "${transactions}" ] ; then
+        echo "$(date): No transactions to recover"
+        success=true
+        break
+      fi
+
+      echo "$(date): Waiting for the following transactions: ${transactions}"
+      sleep ${RECOVERY_PAUSE}
+    done
+
+    if [ "${success}" = "true" ] ; then
+      message="Finished, recovery terminated successfully"
+    else
+      message="Finished, Recovery DID NOT complete, check log for details.  Recovery will be reattempted."
+    fi
+  fi
+
+  if [ ! -f "${terminatingFile}" ] ; then
+      run_cli_cmd ':shutdown' >/dev/null 2>&1
+  fi
+  wait $PID 2>/dev/null
+  trap - TERM
+  wait $PID 2>/dev/null
+
+  echo "$(date): ${message}"
+  if [ "${success}" != "true" ] ; then
+    return 64
+  else
+    return 0
+  fi
+}
+
+STANDALONE_XML=${JBOSS_HOME}/standalone/configuration/${STANDALONE_XML_FILE:-standalone-openshift.xml}
+STANDALONE_XML_COPY=${STANDALONE_XML}.orig
+
+cp -p ${STANDALONE_XML} ${STANDALONE_XML_COPY}
+
+DATA_DIR="${JBOSS_HOME}/standalone/partitioned_data"
```

Comment:
We allow the user to override DATA_DIR in assemble.  I'm not sure how that affects things at runtime, though, since it doesn't look like it's used in any of the launch scripts.  I haven't checked standalone.sh/.conf to see if it's referenced in one of those.  Just FYI.

Reply:
I thought the supported env was APP_DATADIR and this was copied into data, I don't believe the scripts currently allow the user to override DATA_DIR

Reply:
Ah okay, I've found it in the EAP assemble

Reply:
Okay, but if it's not being used during launch, then I don't think it matters.

Reply:
We should add a task to review the use of DATA_DIR, with the split code the assumption is that APP_DATADIR is being used and being copied into standalone/data.  We can make this more flexible and allow DATA_DIR to be overridden.